crypto/asn1/a_int.c

```diff
@@ -319,7 +326,12 @@ static int asn1_string_set_int64(ASN1_STRING *a, int64_t r, int itype)
 
     a->type = itype;
     if (r < 0) {
-        off = asn1_put_uint64(tbuf, -r);
+        /* Most obvious '-r' triggers undefined behaviour for most
+         * common INT64_MIN. Even though below '0 - (uint64_t)r' can
+         * appear two's-complement centric, it does produce correct/
+         * expected result even on one's-complement. This is because
```

Comment:
Well, it's a trick question. While it's indeed correct that we don't care about ones'-complement, we do [and should] strive for making code *formally* correct. I'd argue that recognizing/understanding corner cases like one in question is one of the ways to achieve it.